agent/agent-filtering.md

```diff
@@ -0,0 +1,39 @@
+# Filter interface
+
+The filtering capability allows running an arbitrary filter at the entry point of the application.
+The instrumentation points use the result of the evaluation to block the execution.
+
+## Filter interface
+
+The filter interface defines following evaluation API. Each of these methods
+MAY be called multiple times. If the request headers and body are available at once
+the instrumentation MUST use the API with both entities.
+
+### Evaluate request headers
+
+```java
+boolean evaluateRequestHeaders(Span span, Map<String, String> headers)
+```
+
+### Evaluate request body 
+
+```java
+boolean evaluateRequestBody(Span span, String body)
+```
+
+### Evaluate request headers and body
```

Comment:
Very late to the party. I wonder the reason to increase API surface by introducing this method instead of combining the two filters. Is there any use case for this? cc @pavolloffay @davexroth